homeassistant/components/fronius/config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Config flow for Fronius integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pyfronius import Fronius, FroniusError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_RESOURCE
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, FroniusConfigEntryData
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+
+async def validate_input(
+    hass: HomeAssistant, data: dict[str, Any]
+) -> tuple[str, FroniusConfigEntryData]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    host = data[CONF_HOST]
+    fronius = Fronius(async_get_clientsession(hass), host)
+
+    try:
+        logger_info: dict[str, Any]
+        logger_info = await fronius.current_logger_info()
```

Comment:
The wording is kind of misleading, this looks like a logging object, is this intended?

Reply:
Well the device is called "Fronius Datalogger", the API endpoint `GetLoggerInfo.cgi` and the pyfronius method `current_logger_info` 🤷‍♂️
I could add an additional term like `datalogger_info` but that's also bit confusing imho.

Reply:
I would include data here, that makes it a little less verbose